alica_engine/src/engine/modelmanagement/factories/VariableFactory.cpp

```diff
@@ -14,4 +14,12 @@ Variable* VariableFactory::create(const YAML::Node& variableNode)
 
     return variable;
 }
+
+Variable* VariableFactory::create(int64_t id, const std::string& name, const std::string& type)
+{
+    Variable* variable = new Variable(id, name, type);
```

Comment:
Where is the memory deallocated? I could not find anything in Factory, ModelManager and PlanRepository.

Reply:
it stored in unique_ptr by Factory::storeElement function.

Reply:
Just saw that you wrapped it in a unique_ptr in storeElement now, so memory should be deallocated when the modelManager is destroyed